### DISCUSSION TITLE
invalid opcode: PUSH0 for truffle migrate to linea testnet

### DISCUSSION THREAD

**Discussion by covendev** [ConsenSys-archive/truffle#6067] [Web3 Unleashed]
Hi,
I'm getting the following error while deploying using the following command:
**command**: truffle deploy --network linea --verbose-rpc --interactive --skip-dry-run --reset
**error**:
 <     "id": 12,
 <     "jsonrpc": "2.0",
 <     "error": {
 <       "message": "invalid opcode: PUSH0",
 <       "code": -32000
 <     }
 <   }
 *** Deployment Failed ***

"Token" hit an invalid opcode while deploying. Try:
   * Verifying that your constructor params satisfy all assert conditions.
   * Verifying your constructor code doesn't access an array out of bounds.
   * Adding reason strings to your assert statements.

Exiting: Review successful transactions manually by checking the transaction hashes above on Etherscan.

Error:  *** Deployment Failed ***

"Token" hit an invalid opcode while deploying. Try:
   * Verifying that your constructor params satisfy all assert conditions.
   * Verifying your constructor code doesn't access an array out of bounds.
   * Adding reason strings to your assert statements.

**truffle** **version** : v5.9.2

**Comment by FlaviuWeb3** [comment]
The error message you received indicates that an invalid opcode with the value "PUSH0" was encountered while deploying the "Token" contract using Truffle. This error typically occurs when there is an issue with the contract code or its deployment process. Here are some steps you can take to troubleshoot the problem:

Review constructor parameters: Make sure that the constructor parameters provided for the "Token" contract are valid and satisfy all assert conditions defined within the constructor. Check if there are any specific requirements or constraints mentioned in the contract code that need to be fulfilled during deployment.

Check for array out-of-bounds access: Review the constructor code and ensure that there are no operations that access arrays out of their bounds. Make sure that array indices are properly calculated and do not exceed the array's length.

Add reason strings to assert statements: If there are assert statements in your constructor, consider adding reason strings to them. Reason strings provide additional information about the assertion condition that failed, which can be helpful in identifying the cause of the error. For example, you can modify an assert statement like this: assert(condition, "Assertion failed: invalid value").

Verify Truffle version compatibility: Ensure that the version of Truffle you are using (v5.9.2 in your case) is compatible with the Solidity compiler version specified in your contract. If there is a mismatch between the Truffle version and the Solidity version, it could cause deployment issues. You can check the Solidity version in your contract pragma statement at the beginning of the contract.

Review other error messages: Look for any additional error messages or warnings that might be present in the output. Sometimes, there might be other errors preceding the invalid opcode error that provide more context about the issue.

If none of the above steps resolve the problem, it might be helpful to provide more information about the "Token" contract code or any additional error messages you received.
